pkg/api/validation/schema.go

```diff
@@ -311,6 +311,9 @@ func (s *SwaggerSchema) validateField(value interface{}, fieldName, fieldType st
 		return s.ValidateObject(value, fieldName, fieldType)
 	}
 	allErrs := []error{}
+	if reflect.TypeOf(value) == nil {
```

Comment:
Hi @smarterclayton @kargakis , thanks for the review.

So I had a look at IsZero() in https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/generate.go#L52

The problem with that (as I see it) is it will return true for any of the type specific zeros ("", 0, 0.0) so this means that a user will never be able to set an integer "0" or just a empty string. Is this truly what we want? Wouldn't it *break things*? 

Remember all I am trying to do here is prevent a panic when TypeOf() returns nil

test program:
package main

import (
	"fmt"
	"reflect"
)

func IsZero(i interface{}) bool {
	if i == nil {
		return true
	}
	return reflect.DeepEqual(i, reflect.Zero(reflect.TypeOf(i)).Interface())
}

func validateField(value interface{}, fieldName string) {
	fmt.Printf("%v] value:'%v', IsZero:%v\n", fieldName, reflect.ValueOf(value), IsZero(value))
}

func main() {
	var s *string
	x := "foo"
	e := ""
	i := 9
	var pi *int
	zi := 0

	validateField(s, "*string")
	validateField(x, "valid s")
	validateField(e, "empty s")
	validateField(zi, "0 int  ")
	validateField(i, "valid i")
	validateField(pi, "*int   ")
}

output:

*string] value:'<nil>', IsZero:true
valid s] value:'foo', IsZero:false
empty s] value:'', IsZero:true
0 int  ] value:'0', IsZero:true
valid i] value:'9', IsZero:false
*int   ] value:'<nil>', IsZero:true